### PR TITLE
Implement a SpeechToText provider

### PR DESCRIPTION
**Pull request by julien-nc**
Here is an STT provider using Replicate's Whisper API.
https://replicate.com/openai/whisper/versions/30414ee7c4fffc37e260fcab7842b5be470b9b840f2b608f5baa9bbef9a259ed/api
This one is a bit trickier than the OpenAI's one. We have to wait until the Replicate job is finished.

### CODE REVIEW

File: lib/AppInfo/Application.php [hunk 1]
```diff
@@ -42,6 +42,10 @@ public function register(IRegistrationContext $context): void {
 			$context->registerReferenceProvider(ImageReferenceProvider::class);
 			$context->registerReferenceProvider(WhisperReferenceProvider::class);
 			$context->registerEventListener(RenderReferenceEvent::class, ReplicateReferenceListener::class);
+
+			if (version_compare($this->config->getSystemValueString('version', '0.0.0'), '27.0.0', '>=')) {
```
Comment (marcelklehr) [inline lib/AppInfo/Application.php:46]: Same question whether this is sufficient to still work <27

File: lib/Service/ReplicateAPIService.php [hunk 2]
```diff
@@ -55,6 +58,40 @@ public function createWhisperPrediction(string $audioFileUrl, bool $translate =
...
+						? $prediction['output']['translation']
+						: $prediction['output']['transcription'];
+				} elseif ($prediction['status'] === 'failed') {
+					throw new Exception('Error transcribing file "' . $file->getName() . '": remote job failed');
+				} elseif ($prediction['status'] === 'canceled') {
+					throw new Exception('Error transcribing file "' . $file->getName() . '": remote job was canceled');
+				} elseif ($prediction['status'] !== 'starting' && $prediction['status'] !== 'processing') {
+					throw new Exception('Error transcribing file "' . $file->getName() . '": unknown prediction status');
+				}
+				sleep(2);
```
Comment (marcelklehr) [inline lib/Service/ReplicateAPIService.php:88]: heh, love it!

**Review by marcelklehr — APPROVED** [review]

File: lib/Service/ReplicateAPIService.php (hunk 2 above)
Comment (julien-nc) [reply lib/Service/ReplicateAPIService.php:88]: The `unknown prediction status` message?

Comment (marcelklehr) [reply lib/Service/ReplicateAPIService.php:88]: No, the sleep. We don't get to use sleep often without regretting it :D

**Pull request merged**